gcp/api/server.py

```diff
@@ -340,9 +404,9 @@ def do_query(query, context: grpc.ServicerContext, include_details=True):
     ecosystem = ''
     purl_str = ''
 
-  page_token = None
+  context.page_token = None
   if query.page_token:
-    page_token = ndb.Cursor(urlsafe=query.page_token)
+    context.page_token = ndb.Cursor(urlsafe=query.page_token)
```

Comment:
Does there need to be some validation here or around the queries to prevent an invalid page token causing problems?

Reply:
Not sure how I can validate it, as it's just a base64 encoded byte blob, I think it won't do anything if the bytes does not relate to the query.